include/seqan3/alphabet/composition/union_composition.hpp

```diff
@@ -277,29 +113,57 @@ namespace seqan3
  * assert(letter.to_rank() == 6);
  * ```
  */
-template <typename first_alphabet_type, typename ...alphabet_types>
+template <typename ...alphabet_types>
 //!\cond
-    requires alphabet_concept<first_alphabet_type> && (alphabet_concept<alphabet_types> && ...)
+    requires (alphabet_concept<alphabet_types> && ... && (sizeof...(alphabet_types) >= 1))
 //!\endcond
 class union_composition
 {
+protected:
+    //!\privatesection
+
+    /*!\brief Returns the sum over all alphabet_size_v<alphabets_t>'s.
+     * \tparam ...alphabets_t The types must satisfy seqan3::alphabet_concept.
+     * \hideinitializer
+     *
+     * ```cpp
+     * constexpr auto sum = alphabet_sum_size_v<dna4, gap, dna5>;
+     * assert(sum == 10);
+     * ```
+     */
+    template <typename ...alphabets_t>
+    //!\cond
+        requires (alphabet_concept<alphabets_t> && ...)
+    //!\endcond
+    static constexpr auto alphabet_sum_size_v =
```

Comment:
I think this is okay, because this is a protected function, so it can be technical.

It is a sum of alphabet sizes.

I will rename it into `alphabet_sizes_sum_v`, which might be easier to read.

Reply:
Well, if it is detail then why not being completely expressive?
`sum_of_alphabet_sizes_v` as it is a meta-function that computes something anyway, right?

Reply:
I think this style does not reflect the STL naming scheme. (they don't have _of_ containing in the names)

Reply:
Okay they have (e.g. std::all_of, std::any_of, std::none_of), but they don't use them in the middle part.

I will rename all those meta functions as `sum_of_alphabet_sizes`.

Reply:
[See](http://en.cppreference.com/w/cpp/types/result_of)
I mean it could also be called `sum_up_alphabet_sizes_v`
I figured, that the meta-functions on type_traits are very well using preposition to clearly express the meaning. There is even a mf called `treat_as_floating_point_v`. So I don't see a reason why not expressively naming this mf.

Reply:
renamed. Names are now: `(max|sum|partial_sum)_of_alphabet_sizes`